spdmlib/src/message/measurement.rs

```diff
@@ -98,7 +99,15 @@ pub struct SpdmMeasurementsResponsePayload {
 
 impl SpdmCodec for SpdmMeasurementsResponsePayload {
     fn spdm_encode(&self, context: &mut common::SpdmContext, bytes: &mut Writer) {
-        self.number_of_measurement.encode(bytes); // param1
+        //When Param2 in the requested measurement operation is 0 , this
+        //parameter shall return the total number of measurement indices on
+        //the device. Otherwise, this field is reserved.
+        if self.measurement_record.number_of_blocks == 0 {
+            // SpdmMeasurementOperation::SpdmMeasurementQueryTotalNumber
+            (MAX_SPDM_MEASUREMENT_BLOCK_COUNT as u8).encode(bytes); // param1
```

Comment:
This is wrong.
We cannot use MAX_SPDM_MEASUREMENT_BLOCK_COUNT.
We must use real measurement block count from device security lib.

Reply:
Fine, But I have no idea where could I get the value from. Or just use 5 now?

Reply:
spdmlib should get it from below function. No hardcode is allowed in spdmlib.

type SpdmMeasurementCollectionCbType = fn(
    spdm_version: SpdmVersion,
    measurement_specification: SpdmMeasurementSpecification,
    measurement_hash_algo: SpdmBaseHashAlgo,
    measurement_index: usize,
) -> Option<SpdmMeasurementRecordStructure>;